framework/web/Request.php

```diff
@@ -842,8 +950,8 @@ public function setUrl($value)
      */
     protected function resolveRequestUri()
     {
-        if (isset($_SERVER['HTTP_X_REWRITE_URL'])) { // IIS
-            $requestUri = $_SERVER['HTTP_X_REWRITE_URL'];
+        if ($this->headers->has('X-Rewrite-Url')) { // IIS
```

Comment:
It's hard to find any documentation, but I think `X-Rewrite-Url` is not even part of IIS, but of a 3rd party plugin: http://www.helicontech.com/isapi_rewrite/

https://forums.asp.net/post/5807236.aspx

This could be resolved in a separate issue as well. I'll make one.